### PR TITLE
Updates to LiteLLM integration

### LINKED ISSUE #9472: [Bug]: LiteLLM integration issues when querying
### Short description of current behavior

When querying the LiteLLM integration, the following error is returned:
```
[litellm/litellm_handler_simple1]: BadRequestError: litellm.BadRequestError: BadRequestError: OpenAIException - Error code: 400 - {'error': {'message': "Invalid 'functions': empty array. Expected an array with minimum length 1, but got an empty array instead.", 'type': 'invalid_request_error', 'param': 'functions', 'code': 'empty_array'}}
```
Steps to reproduce:
1. Create new model with litellm engine:
```
create model litellm_handler_simple6
predict text
using
engine="litellm",
model="gpt-4",
api_key="sk-xxx";
```
2. Query the above model:
```
select * from litellm_handler_simple6 where text='Once upon a time';
```

I guess this is related to sending few deprecated parameters to the latest OpenAI API 

### Video or screenshots

_No response_

### Expected behavior

_No response_

### How to reproduce the error

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ZoranPandovski**
## Description

This PR fixes #9472 , https://github.com/mindsdb/mindsdb/security/dependabot/142 and https://github.com/mindsdb/mindsdb/security/dependabot/143 and also updates the README to follow up to the new structure.

## Type of change

(Please delete options that are not relevant)

- [X] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [X] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/litellm_handler/README.md [hunk 1]
```diff
@@ -1,120 +1,60 @@
-# Litellm Handler
+## LlamaIndex Handler
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/litellm_handler/README.md:1]: This should be LiteLLM.

File: mindsdb/integrations/handlers/litellm_handler/README.md [hunk 2]
```diff
@@ -1,120 +1,60 @@
...
-api_key="openai-api-key",
-prompt_template="write a story based on {{text}}";
+## Usage
 
-select * from information_schema.models where name ="litellm_handler_prompt_template" ;
+Here is how to create a model that answers questions:
 
-select * from litellm_handler_prompt_template where text='Once upon a time';
+```sql
+CREATE MOEL chat_model
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/litellm_handler/README.md:28]: This should be `CREATE MODEL` (the 'D' is missing).

File: mindsdb/integrations/handlers/litellm_handler/settings.py [hunk 3]
```diff
@@ -6,9 +6,8 @@ class CompletionParameters(BaseModel):
     model: str  # The model to be used for the API call.
     prompt_template: Optional[str] = None  # Template for the prompt to be used.
 
-    # Optional OpenAI params: see https://platform.openai.com/docs/api-reference/chat/create
-    functions: List = []  # List of functions for advanced model operations.
-    function_call: str = ""  # String to specify a particular function call, if needed.
+    # Optional OpenAI params: see  https://platform.openai.com/docs/api-reference/runs
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/litellm_handler/settings.py:9]: This link refers to the Assistants API, but I think we are calling the Chat Completions API within the handler. Should it be changed?

File: mindsdb/integrations/handlers/litellm_handler/README.md [hunk 4]
```diff
@@ -1,120 +1,60 @@
...
-create model litellm_handler_messages
-predict text
-using
-engine="litellm",
-model="gpt-3.5-turbo",
-api_key="openai-api-key";
-       
-select * from information_schema.models where name ="litellm_handler_messages" ;
+**Next Steps**
+Go to the [Use Cases](https://docs.mindsdb.com/use-cases/overview) section to see more examples.
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/litellm_handler/README.md:58]: As far as I am aware, we don't have any tutorials using LiteLLM; should we still include this here?

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]
Hey @ZoranPandovski,
I left a few minor comments on the changes that have been made,

One other thing I noticed when looking at the handler code is that there are couple of syntax errors [here](https://github.com/mindsdb/mindsdb/blob/95e013786af391d951eb076f63d336ebd35d34a7/mindsdb/integrations/handlers/litellm_handler/litellm_handler.py#L113) and [here](https://github.com/mindsdb/mindsdb/blob/95e013786af391d951eb076f63d336ebd35d34a7/mindsdb/integrations/handlers/litellm_handler/litellm_handler.py#L125). Should we fix these?

The typical flow, i.e., the example given in the README seems to avoid these statements due to the conditions they are bound by.

File: mindsdb/integrations/handlers/litellm_handler/README.md (hunk 4 above)
Comment (ZoranPandovski) [reply mindsdb/integrations/handlers/litellm_handler/README.md:58]: We are using this as a general go to section

**Review by MinuraPunchihewa — APPROVED** [review]

**Pull request merged**